### PR TITLE
Adds batched option to quest's InOutOctree containment example

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR is a feature
- It adds a ``--batched`` flag to the quest::InOutOctree point containment example
- When this is set and the code is built using RAJA and OpenMP, it will run the containment query
over an array of points using the RAJA OMP execution policy.

### CODE REVIEW

**Comment by kennyweiss** [comment]
I'm seeing about a 10x speedup in batched mode with OpenMP on a toss3 node with 36 cores:

Release mode build without batched:
```
>./examples/quest_containment_driver_ex ../data/quest/plane.stl    
[INFO] ******************************* Loading the mesh ******************************* 
[INFO] Reading file: ../data/quest/plane.stl... 
[INFO] Mesh has 1120242 nodes and 373414 cells. 
...
[INFO] ***************************** Querying the octree ****************************** 
[INFO] Bounding box for query points: { min:(-419.598,-36.1974,-33.8927); max:(419.598,845.432,198.783); range:<839.196,881.629,232.675> } 
[INFO] 	Querying (2,2,2)^3 containment field took 8.4e-06 seconds (@ 952381 queries per second) 
[INFO] 	Querying (4,4,4)^3 containment field took 2.6735e-05 seconds (@ 2.39387e+06 queries per second) 
[INFO] 	Querying (8,8,8)^3 containment field took 0.000208795 seconds (@ 2.45217e+06 queries per second) 
[INFO] 	Querying (16,16,16)^3 containment field took 0.00113226 seconds (@ 3.61754e+06 queries per second) 
[INFO] 	Querying (32,32,32)^3 containment field took 0.0055574 seconds (@ 5.89628e+06 queries per second) 
[INFO] 	Querying (64,64,64)^3 containment field took 0.041109 seconds (@ 6.3768e+06 queries per second) 
[INFO] 	Querying (128,128,128)^3 containment field took 0.302812 seconds (@ 6.9256e+06 queries per second) 
[INFO] 	Querying (256,256,256)^3 containment field took 2.1179 seconds (@ 7.92161e+06 queries per second) 
```
Release mode build in batched mode:
```
>./examples/quest_containment_driver_ex ../data/quest/plane.stl --batched
[INFO] ******************************* Loading the mesh ******************************* 
[INFO] Reading file: ../data/quest/plane.stl... 
[INFO] Mesh has 1120242 nodes and 373414 cells. 
...
[INFO] ***************************** Querying the octree ****************************** 
[INFO] Bounding box for query points: { min:(-419.598,-36.1974,-33.8927); max:(419.598,845.432,198.783); range:<839.196,881.629,232.675> } 
[INFO] 	Querying (2,2,2)^3 containment field took 0.0353549 seconds (@ 226.277 queries per second) 
[INFO] 	Querying (4,4,4)^3 containment field took 0.000484981 seconds (@ 131964 queries per second) 
[INFO] 	Querying (8,8,8)^3 containment field took 0.000661396 seconds (@ 774120 queries per second) 
[INFO] 	Querying (16,16,16)^3 containment field took 0.00121585 seconds (@ 3.36884e+06 queries per second) 
[INFO] 	Querying (32,32,32)^3 containment field took 0.00794509 seconds (@ 4.12431e+06 queries per second) 
[INFO] 	Querying (64,64,64)^3 containment field took 0.0184896 seconds (@ 1.41779e+07 queries per second) 
[INFO] 	Querying (128,128,128)^3 containment field took 0.049532 seconds (@ 4.23393e+07 queries per second) 
[INFO] 	Querying (256,256,256)^3 containment field took 0.271726 seconds (@ 6.1743e+07 queries per second) 
```

Tag: @adayton1

**Comment by kennyweiss** [comment]
Passed all `toss3` and `blueos` host-configs.

File: src/axom/primal/tests/primal_numeric_array.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ TEST(primal_numeric_array, constructors)
     EXPECT_EQ(0, fromInitializerListTooShort[i]);
   }
 
-  primal::NumericArray<int, 3> fromInitializerNoEquaslSign{10, 20, 30};
+  primal::NumericArray<int, 3> fromInitializerNoEquaslSign {10, 20, 30};
```
Comment (agcapps) [inline src/axom/primal/tests/primal_numeric_array.cpp:83]: `NoEquaslSign` ?

File: src/axom/primal/tests/primal_vector.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ TEST(primal_vector, vector_constructors)
     EXPECT_EQ(0, fromInitializerListTooShort[i]);
   }
 
-  primal::Vector<int, 3> fromInitializerNoEquaslSign{10, 20, 30};
+  primal::Vector<int, 3> fromInitializerNoEquaslSign {10, 20, 30};
```
Comment (agcapps) [inline src/axom/primal/tests/primal_vector.cpp:90]: Oh no!! the typo strikes again.

**Review by agcapps — APPROVED** [review]
Thanks, @kennyweiss .  The speed-ups are gratifying.  Pending the typo fix, approved.

File: src/axom/primal/tests/primal_numeric_array.cpp (hunk 1 above)
Comment (kennyweiss) [reply src/axom/primal/tests/primal_numeric_array.cpp]: Good catch!  Fixed.

File: src/axom/primal/tests/primal_vector.cpp (hunk 2 above)
Comment (kennyweiss) [reply src/axom/primal/tests/primal_vector.cpp]: Fixed.

**Review by rhornung67 — APPROVED** [review]
Nice! 🥇

**Pull request merged**